### PR TITLE
Update readme files

### PR DESCRIPTION
**Pull request by bfriesen**
- Create separate readme files for each package.
- Update main readme to link to each package.
- Remove old content from main readme.

### CODE REVIEW

**Comment by derikwhittaker** [comment]
I like the build status.  Smart on breaking into smaller files

File: README.md [hunk 1]
```diff
@@ -1,185 +1,13 @@
...
-- RockLib.Configuration
-  - [![Build status](https://ci.appveyor.com/api/projects/status/0qxs1k1bw36cn8ly?svg=true)](https://ci.appveyor.com/project/bfriesen/rocklib-configuration)
-- RockLib.Configuration.ObjectFactory
-  - [![Build status](https://ci.appveyor.com/api/projects/status/ox9velgud5ljj8d0?svg=true)](https://ci.appveyor.com/project/bfriesen/rocklib-configuration-owv0n)
-- RockLib.Configuration.ProxyFactory
-  - [![Build status](https://ci.appveyor.com/api/projects/status/ogf4axfokbklh638?svg=true)](https://ci.appveyor.com/project/bfriesen/rocklib-configuration-fe27m)
+### [RockLib.Configuration.ObjectFactory](RockLib.Configuration.ObjectFactory) [![Build status](https://ci.appveyor.com/api/projects/status/ox9velgud5ljj8d0?svg=true)](https://ci.appveyor.com/project/bfriesen/rocklib-configuration-owv0n)
 
-All RockLib packages will depend on the RockLib.Configuration package to provide its configuration features.
+An alternative to `Microsoft.Extensions.Configuration.Binder` that supports readonly properties and other features commonly found in JSON and XML serializers.
```
Comment (JustinMoss) [inline README.md]: This line does not exactly match the summary for the RockLib.Configuration.ObjectFactory.  Though it's not wrong, it's not consistent with the other two packages summaries.

File: RockLib.Configuration.ObjectFactory/README.md [hunk 2]
```diff
@@ -0,0 +1,392 @@
...
+    }
+
+    public class BazDerived : Baz
+    {
+        public bool Corge { get; set; }
+    }
+}
+```
+
+a instance of `Foo` can be created with a `Baz` property with a value of type `BazDerived` by specifying a `type/value` pair where the `type` is a *[assembly quallified name](https://msdn.microsoft.com/en-us/library/system.type.assemblyqualifiedname.aspx)*:
```
Comment (JustinMoss) [inline RockLib.Configuration.ObjectFactory/README.md]: [assembly quallified name] should be [assembly qualified name]

File: RockLib.Configuration.ObjectFactory/README.md [hunk 3]
```diff
@@ -0,0 +1,392 @@
...
+public interface IBar
+{
+}
+
+public class DefaultBar : IBar
+{
+    public int Baz { get; set; }
+}
+
+We would like to be able to define the `DefaultBar` type as the default type - if the type is not otherwise specified, create an instance of `DefaultBar`. There are two ways of specifying the default type: by property or by type. When specified by property, we want to say: "when setting the `Foo.Bar` property, if the configuration is *not* type-specified, set the property to an instance of `DefaultBar`." When specified by type we want to say: "whenever creating an instance of `IBar`, if the configuration is *not* type-specified, create an instance of `DefaultBar`.
```
Comment (JustinMoss) [inline RockLib.Configuration.ObjectFactory/README.md]: Should this close off the code section?  It seems to look funny here in the readme

File: RockLib.Configuration.ObjectFactory/README.md [hunk 4]
```diff
@@ -0,0 +1,392 @@
...
+
+public class Baz : IBaz
+{
+    public bool Corge { get; set; }
+}
+```
+
+### Value Converters
+
+RockLib converts most configuration string values to the target type by using the `TypeConverter` obtained by calling `TypeDescriptor.GetConverter(targetType)`. In addition, there is support for target types `Encoding` and `Type`. If value conversions need to be supported for additional types, value converters can be registered. These value converters a functions that have a single string parameter and return the target type. Similar to default types, they can be registered by property or by type.
```
Comment (JustinMoss) [inline RockLib.Configuration.ObjectFactory/README.md]: "These value converters a functions" should be "These value converters are functions"

**Review by JustinMoss — CHANGES_REQUESTED** [review]
Overall great.  Found a few typos I think though.

**Review by JustinMoss — APPROVED** [review]

**Pull request merged**